src/main/java/controller/LottoGameController.java

```diff
@@ -0,0 +1,89 @@
+import domain.result.LottoRank;
+import domain.result.LottoResult;
+import domain.result.WinningResult;
+import view.InputView;
+import view.OutputView;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class LottoGameController {
+    public void run() {
+        final GameMoney gameMoney = makeGameMoney();
+
+        final LottoBundle lottoBundle = makeLottoBundle(gameMoney);
+
+        final WinningResult winningResult = makeWinningResult();
+
+        makeLottoResult(lottoBundle, winningResult, gameMoney);
+    }
+
+    private GameMoney makeGameMoney() {
+        try {
+            OutputView.printGameMoneyRequest();
+            int userGameMoney = InputView.getGameMoney();
+            return new GameMoney(userGameMoney);
+        } catch (IllegalArgumentException e) {
+            OutputView.printErrorMessage(e.getMessage());
+            return makeGameMoney();
+        }
+    }
+
+    private LottoBundle makeLottoBundle(final GameMoney gameMoney) {
+        final LottoBundle lottoBundle = gameMoney.buyLotto();
+        OutputView.printLottoBought(lottoBundle);
+        return lottoBundle;
+    }
+
+    private WinningResult makeWinningResult() {
+        try {
+            final Lotto winningLotto = makeWinningLotto();
+            final LottoBall bonusBall = makeBonusBall();
+            return new WinningResult(winningLotto, bonusBall);
+        } catch (IllegalArgumentException e) {
+            OutputView.printErrorMessage(e.getMessage());
+            return makeWinningResult();
+        }
+    }
+
+    private Lotto makeWinningLotto() {
+        try {
+            OutputView.printWinningLottoRequest();
+            final List<Integer> winningLottoNumber = InputView.getWinningLotto();
+            return new Lotto(winningLottoNumber.stream()
+                    .map(number -> LottoBall.valueOf(number))
+                    .collect(Collectors.toList()));
+        } catch (IllegalArgumentException e) {
+            OutputView.printErrorMessage(e.getMessage());
+            return makeWinningLotto();
+        }
+    }
+
+    private LottoBall makeBonusBall() {
+        try {
+            OutputView.printBonusBallRequest();
+            final int bonusBall = InputView.getBonusBall();
+            return LottoBall.valueOf(bonusBall);
+        } catch (IllegalArgumentException e) {
+            OutputView.printErrorMessage(e.getMessage());
+            return makeBonusBall();
+        }
+    }
+
+    private void makeLottoResult(final LottoBundle lottoBundle, final WinningResult winningResult, final GameMoney gameMoney) {
+        final LottoResult lottoResult = new LottoResult();
+
+        final Map<LottoRank, Integer> gameResult = lottoResult.checkResult(lottoBundle, winningResult);
+        OutputView.printLottoResult(gameResult);
+
+        final double profitRate = lottoResult.checkProfitRate(gameResult);
+        OutputView.printProfitRate(profitRate);
```

Comment:
```suggestion
        final LottoResult lottoResult = lottoBundle.checkResult(winningResult);
        OutputView.printLottoResult(lottoResult);

        final double profitRate = lottoResult.getProfitRate();
        OutputView.printProfitRate(profitRate);
```

객체에게 역할을 넘겨주고 메시지를 주고받는 구조로 만들어보면 좋겠습니다!
위 코드는 전체적인 느낌을 주기 위해 간단하게 작성한거라 어떠한 구조인지만 참고해주세요!

Reply:
아 네네! LottoBundle에게 그냥 구매한 로또들을 들고있는 것 뿐만아니라 결과를 확인하도록 하는 책임도 쥐여줄수 있겠네요. 리팩토링 하면서 적용해보겠습니다 :)